tests/unit_tests/CMakeLists.txt

```diff
@@ -25,6 +25,11 @@ if("lapack" IN_LIST TARGET_DOMAINS)
     find_package(LAPACKE REQUIRED)
 endif()
 
+# Setup config header with path to reference libraries
+set(ONEMKL_TESTS_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/config.hpp.in
+               ${ONEMKL_TESTS_BINARY_DIR}/include/config.hpp)
```

Comment:
I'm not keen to have the same name `config.hpp`. I would suggest to either rename it to `test_config.hpp` for instance or even better IMO: do not use a config file and instead directly define `REF_BLAS_LIBNAME` and `REF_CBLAS_LIBNAME` as macros when compiling the tests. This could be done in https://github.com/oneapi-src/oneMKL/blob/develop/tests/unit_tests/CMakeLists.txt when the domain is "blas" or "lapack" using `target_compile_definitions`.
I've just never been a fan of generated headers.